src/overlays/actors/ovl_Obj_Warp2block/z_obj_warp2block.c

```diff
@@ -43,36 +42,305 @@ const ActorInit Obj_Warp2block_InitVars = {
+        DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+    }
+}
+
+void ObjWarp2block_SetInactive(ObjWarp2block *this) {
+    this->actionFunc_164 = ObjWarp2block_DoNothing;
+    this->dyna.actor.draw = NULL;
+}
+
+void ObjWarp2block_DoNothing(ObjWarp2block *this, GlobalContext *globalCtx) {}
+
+void func_80BA24E8(ObjWarp2block *this) {
+    this->actionFunc_164 = func_80BA24F8;
+}
+
+void func_80BA24F8(ObjWarp2block *this, GlobalContext *globalCtx) {
+    Actor *current;
+
+    current = globalCtx->actorCtx.actorLists[7].head;
+
+    while (current != NULL) {
+        if (current->id == 0x1D6) {
+            if (((current->params >> 0xF) & 1) == 0) {
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Warp2block/func_80BA1ECC.s")
+                if ((this->dyna.actor.params & 0x3F) == (current->params & 0x3F)) {
+                    this->dyna.actor.child = current;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Warp2block/func_80BA2048.s")
+                    if (Flags_GetSwitch(globalCtx, this->dyna.actor.params & 0x3F) != 0) {
+                        ObjWarp2block_SwapWithChild(this, globalCtx);
+                        this->dyna.actor.draw = ObjWarp2block_Draw;
+                    }
+                    func_80BA2600(this);
+                    return;
+                }
+            }
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Warp2block/func_80BA2218.s")
+        current = current->next;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Warp2block/func_80BA228C.s")
+    this->unk_174 = this->unk_174 + 1;
+    if (this->unk_174 >= 0x3D) {
+        osSyncPrintf("\x1b[41;37m");
+        osSyncPrintf("Error : 時のブロック(ワープ２)が対でセットされていません(%s %d)\n", "../z_obj_warp2block.c", 0x1F9);
+        osSyncPrintf("\x1b[m");
+        Actor_Kill(&this->dyna.actor);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Warp2block/func_80BA2304.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Warp2block/ObjWarp2block_Init.s")
+void func_80BA2600(ObjWarp2block *this) {
+    this->actionFunc_164 = func_80BA2610;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Warp2block/ObjWarp2block_Destroy.s")
+void func_80BA2610(ObjWarp2block *this, GlobalContext *globalCtx) {
+    s16 temp;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Warp2block/func_80BA24C8.s")
+    if ((func_80BA2304(this, globalCtx) != 0) && ((s32) this->unk_16C <= 0)) {
+        ObjWarp2block_Spawn(this, globalCtx);
+        this->unk_16C = 0xA0;
+        func_80080480(globalCtx, (Actor *) this);
+        this->unk_170 = 0xC;
+    }
+    temp = this->unk_170;
+    if (temp > 0) {
+        this->unk_170 = temp - 1;
+        if (this->unk_170 == 0) {
+            ObjWarp2block_SwapWithChild(this, globalCtx);
+        }
+    }
+    if (this->unk_16C == 0x32) {
+        func_80078884((u16)0x4807U);
```

Comment:
```suggestion
        func_80078884(0x4807);
```
This is a SFX macro.